index.ts

```diff
@@ -386,6 +386,16 @@ export class HttpClient {
         // which will leak the open socket.
         await response.readBody()
 
+        // strip authorization header if redirected to a different hostname
```

Comment:
Note that I comments out these lines and confirmed tests failed.  Then uncommented and both pass